code/hud/hudets.cpp

```diff
@@ -174,6 +159,34 @@ float ets_get_max_speed(object* objp, float engine_energy)
 	}
 }
 
+void update_max_speed(object* ship_objp)
+{
+	// get max possible speed then adjust and set based on engine health
```

Comment:
Need some asserts and early returns so that if someone calls this function incorrectly in the future, we won't be left with a mystery ctd.

Reply:
Thanks for kicking off the review. The `ets_get_max_speed` function above uses these checks, so I used that format in this new push.

Reply:
I mean that's all I saw.  It's not a large pr, and it makes sense.  Standardizing how a value is set should logocally fix the issue and prevent future bugs as well.